### DISCUSSION TITLE
Calling root_validator on pydantic body input yields dictionary type

### DISCUSSION THREAD

**Discussion by moldhouse** [fastapi/fastapi#7305] [Questions] [answered]
### First check

* [ x ] I added a very descriptive title to this issue.
* [ x ] I used the GitHub search to find a similar issue and didn't find it.
* [ x ] I searched the FastAPI documentation, with the integrated search.
* [ x ] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [ x ] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ x ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ x ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ x ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ x ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI
from pydantic import BaseModel, root_validator
from typing import Optional, Dict, Any

app = FastAPI()

class UserUpdate(BaseModel):
    password: Optional[str]
    name: Optional[str]

    @root_validator
    def validate(cls, values: Dict[str, Any]) -> Dict[str, Any]:
        return values

@app.patch("/me", response_model=schemas.User)
def update_user_me(
    user_in: UserUpdate,
) -> Any:
    assert isinstance(user_in, dict)
    return user_in
```

### Description

* Using root_validator in the pydantic model changes the behavior of the dependency injection, instead of return an instance of UserUpdate, a dict is returned
* Commenting out the root_validator in the pydantic model leads to the expected behavior, in this case user_in is an instance of UserUpdate
* I would expect user_in to be of type UserUpdate

### Environment

* OS: [e.g. Linux / Windows / macOS]: Linux
* FastAPI Version [e.g. 0.3.0]: 0.61.1

* Python version: 3.8.5

**Comment by David-Lor** [comment, marked as answer]
You should rename your "def validate" to something else (like "def _validate"). "validate" is an existing method for pydantic's BaseModel, so you are overriding it with yours, causing that unexpected behaviour. I tried it and after renaming the root_validator function name, it works as expected.

**Comment by tiangolo** [comment]
Thanks for the help @David-Lor! ☕ 

And thanks for closing the issue @moldhouse 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by Andrew-Chen-Wang** [comment]
For SEO purposes since this took me forever to find this issue, here was my traceback:

```
    data = data.dict(exclude_none=True, exclude_unset=True)
AttributeError: 'dict' object has no attribute 'dict'
```
